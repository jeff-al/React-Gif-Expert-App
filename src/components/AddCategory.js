import { useState } from "react"
import PropTypes from 'prop-types'

const AddCategory = ( { setCategories } ) => {

    const [value, setInputValue] = useState("");

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(value.trim() !== ""){
            setCategories( categories => [value, ...categories])
            setInputValue("");
        }
    }

    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input 
                    type="text"
                    placeholder="Agregar CategorÃ­a"
                    value={ value }
                    onChange={ handleInputChange }
                ></input>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
