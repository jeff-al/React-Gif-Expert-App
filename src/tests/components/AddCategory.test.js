import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("AddCategory tests", () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/>);
    });

    test('should be showed correctly', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('should change input', () => {
        
        const input = wrapper.find("input");
        const value = "Hola mundo";

        input.simulate("change", { target: { value } });

        const p = wrapper.find("p");

        expect(p.text().trim()).toBe(value);

    });

    test('should not post information on submit', () => {
        
        const form = wrapper.find("form");

        form.simulate("submit", { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('should call setCategories and clear input', () => {
        
        const value = "Test";
        const input = wrapper.find("input");

        input.simulate("change", { target: { value } });
        let p = wrapper.find("p");

        expect( p.text().trim() ).toBe(value);

        const form = wrapper.find("form");

        form.simulate("submit", { preventDefault(){} });
        p = wrapper.find("p");

        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( p.text().trim() ).toBe("");

    });
    

})