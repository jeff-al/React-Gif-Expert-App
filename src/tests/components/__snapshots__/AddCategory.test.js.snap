// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests should be showed correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      placeholder="Agregar CategorÃ­a"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
